dirs/dirs.go

```diff
@@ -220,34 +220,41 @@ func SnapHomeDirs() []string {
 func SetSnapHomeDirs(homedirs string) []string {
 	snapHomeDirsMu.Lock()
 	defer snapHomeDirsMu.Unlock()
+
+	//clear old values
+	snapHomeDirs = nil
+	SnapDataHomeGlob = nil
+	HiddenSnapDataHomeGlob = nil
+
+	// Do not set the root directory as home unless explicitly specified with "."
 	if homedirs != "" {
 		snapHomeDirs = strings.Split(homedirs, ",")
 		for i := range snapHomeDirs {
+			// Make the path absolute if relative, also cleans the path for unit tests and HasPrefix
+			snapHomeDirs[i] = filepath.Join("/", snapHomeDirs[i])
```

Comment:
this look strange we should check or fail elsewhere if those values are relative